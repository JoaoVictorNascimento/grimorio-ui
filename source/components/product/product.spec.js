import '../../../internals/test/helper';

import Product from './product-component';
import Panel from '../panel/panel-component';

const exampleProduct = {
  img: 'https://images-americanas.b2w.io/produtos/01/00/sku/33446/6/33446652_4GG.jpg',
  name: 'Notebook Profissional Avell W155 MX Intel Core i7 16GB (GeForce MX150) 1TB 15.6 FullHD',
  info: {
    value: 5333.20,
  },
  expires: '2018-07-08 23:59',
  link: 'https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd',
  tags: [
    {
      type: 'brand',
      value: 'acom',
    },
    {
      type: 'highlight',
      value: true,
    }
  ],
};

const bigInfoProd = {
  ...exampleProduct,
  info: {
    value: 'UMCUPOMBEMGRANDE',
    rules: 'Confira as regras no site',
  },
};

const noImgProd = {
  ...exampleProduct,
  img: null,
};

const noInfoProd = {
  ...exampleProduct,
  info: null,
};

const props = {
  onCopy: jest.fn(),
  onGenerate: jest.fn(),
  stage: 'generate',
};

describe('Product component', () => {

  describe('Product stage: generate', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = mount(
        <Product {...props} data={exampleProduct} />
      );
    });

    it('Render correctly', () => {
      expect(wrapper.find('.imgDefault').length).toEqual(0);
      expect(wrapper.debug()).toMatchSnapshot();
    });

    it('onGenerate', () => {
        jest.spyOn(props, 'onGenerate')

        wrapper.find('button').simulate('click');
        expect(props.onGenerate).toHaveBeenCalled();
     });
  });

  describe('Variations', () => {
    it('Product Info isBig', () => {
      const wrapper = shallow(
        <Product data={bigInfoProd} />
      );

      expect(wrapper.find('.info.isBig').length).toEqual(1);
      expect(wrapper.debug()).toMatchSnapshot();
    });

    it('Type card renders Panel', () => {
      const wrapper = shallow(
        <Product type="card" brand="acom" data={exampleProduct} />
      );
      expect(wrapper.find(Panel).length).toEqual(1);
      expect(wrapper.debug()).toMatchSnapshot();
    });

    it('Product has no image, shows default img', () => {
      const wrapper = shallow(
        <Product data={noImgProd} />
      );

      expect(wrapper.find('.imgDefault').length).toEqual(1);
      expect(wrapper.debug()).toMatchSnapshot();
    });

    it('Product with no info', () => {
      const wrapper = shallow(
        <Product data={noInfoProd} />
      );

      expect(wrapper.find('.info').length).toEqual(0);
      expect(wrapper.debug()).toMatchSnapshot();
    });

  });

  describe('#doesnt render', () => {
    it('doesnt render if doenst receive data props', () => {
      const wrapper = shallow(
        <Product />
      );
      expect(wrapper.type()).toEqual(null);
    });
  });
});
