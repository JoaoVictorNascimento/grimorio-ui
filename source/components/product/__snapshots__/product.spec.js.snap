// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component Product stage: generate Render correctly 1`] = `
"<Product onCopy={[Function: mockConstructor]} onGenerate={[Function: mockConstructor]} stage=\\"generate\\" data={{...}} type=\\"default\\" btnText=\\"Copiar Link\\" generateLoading={false} copyValue={{...}}>
  <section className=\\"wrapper default\\" onCopy={[Function: mockConstructor]} stage=\\"generate\\" data={{...}}>
    <div className=\\"tag\\">
      <Svg width={48} height={48} src=\\"logo/acom\\" align=\\"middle\\" />
      <Svg className=\\"tagHighlight\\" width={32} height={32} src=\\"flame\\" align=\\"middle\\" />
    </div>
    <div className=\\"imgWrapper\\">
      <a target=\\"_blank\\" href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
        <img className=\\"imgCustom\\" src=\\"https://images-americanas.b2w.io/produtos/01/00/sku/33446/6/33446652_4GG.jpg\\" alt=\\"Notebook Profissional Avell W155 MX Intel Core i7 16GB (GeForce MX150) 1TB 15.6 FullHD\\" />
      </a>
    </div>
    <h1 className=\\"name\\">
      <a href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
        Notebook Profissional Avell W155 MX Intel Core i7 16GB (GeForce MX150) 1TB 15.6 FullHD
      </a>
    </h1>
    <div className=\\"info\\">
      <a target=\\"_blank\\" href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
        R$ 5.333,20
      </a>
    </div>
    <div className=\\"footer\\">
      <div className=\\"expires\\">
        <a target=\\"_blank\\" href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
          Valido até: 09/07/2018 2:59
        </a>
      </div>
      <div className=\\"social\\">
        <Button className=\\"copy\\" onClick={[Function: mockConstructor]} loading={false} iconRight=\\"insert_link\\" color=\\"primary\\" modifier=\\"outline\\" disabled={false} size=\\"medium\\" type=\\"button\\" active={false}>
          <button type=\\"button\\" className=\\"copy primary outline medium\\" onClick={[Function: mockConstructor]} disabled={false}>
            Gerar Link
            <Icon className=\\"iconRight\\" size={15} name=\\"insert_link\\" align=\\"middle\\">
              <i style={{...}} className=\\"material-icons icon iconRight\\" align=\\"middle\\">
                insert_link
              </i>
            </Icon>
          </button>
        </Button>
        <Svg onClick={[Function]} className=\\"facebook\\" align=\\"top\\" width={26} height={26} src=\\"icon/facebook-square\\" />
        <Svg onClick={[Function]} className=\\"twitter\\" align=\\"top\\" width={26} height={26} src=\\"icon/twitter-square\\" />
      </div>
    </div>
  </section>
</Product>"
`;

exports[`Product component Variations Product Info isBig 1`] = `
"<section className=\\"wrapper default\\" data={{...}} stage=\\"generate\\" onCopy={[Function: onCopy]}>
  <Symbol(react.fragment)>
    <div className=\\"tag\\">
      <Svg width={48} height={48} src=\\"logo/acom\\" align=\\"middle\\" />
      <Svg className=\\"tagHighlight\\" width={32} height={32} src=\\"flame\\" align=\\"middle\\" />
    </div>
    <div className=\\"imgWrapper\\">
      <a target=\\"_blank\\" href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
        <img className=\\"imgCustom\\" src=\\"https://images-americanas.b2w.io/produtos/01/00/sku/33446/6/33446652_4GG.jpg\\" alt=\\"Notebook Profissional Avell W155 MX Intel Core i7 16GB (GeForce MX150) 1TB 15.6 FullHD\\" />
      </a>
    </div>
    <h1 className=\\"name\\">
      <a href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
        Notebook Profissional Avell W155 MX Intel Core i7 16GB (GeForce MX150) 1TB 15.6 FullHD
      </a>
    </h1>
    <div className=\\"info isBig\\">
      <a target=\\"_blank\\" href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
        UMCUPOMBEMGRANDE
      </a>
      <Tooltip className=\\"rules\\" width=\\"220px\\" message=\\"Confira as regras no site\\" position=\\"top\\" icon=\\"info\\">
        <span className=\\"rulesIcon\\">
          ?
        </span>
      </Tooltip>
    </div>
    <div className=\\"footer\\">
      <div className=\\"expires\\">
        <a target=\\"_blank\\" href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
          Valido até: 09/07/2018 2:59
        </a>
      </div>
      <div className=\\"social\\">
        <Button className=\\"copy\\" onClick={[undefined]} loading={false} iconRight=\\"insert_link\\" color=\\"primary\\" modifier=\\"outline\\" disabled={false} size=\\"medium\\" type=\\"button\\" active={false}>
          Gerar Link
        </Button>
        <Svg onClick={[Function]} className=\\"facebook\\" align=\\"top\\" width={26} height={26} src=\\"icon/facebook-square\\" />
        <Svg onClick={[Function]} className=\\"twitter\\" align=\\"top\\" width={26} height={26} src=\\"icon/twitter-square\\" />
      </div>
    </div>
  </Symbol(react.fragment)>
</section>"
`;

exports[`Product component Variations Product has no image, shows default img 1`] = `
"<section className=\\"wrapper default\\" data={{...}} stage=\\"generate\\" onCopy={[Function: onCopy]}>
  <Symbol(react.fragment)>
    <div className=\\"tag\\">
      <Svg width={48} height={48} src=\\"logo/acom\\" align=\\"middle\\" />
      <Svg className=\\"tagHighlight\\" width={32} height={32} src=\\"flame\\" align=\\"middle\\" />
    </div>
    <div className=\\"imgWrapper\\">
      <a target=\\"_blank\\" href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
        <Svg className=\\"imgDefault\\" src=\\"cupom\\" align=\\"middle\\" />
      </a>
    </div>
    <h1 className=\\"name\\">
      <a href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
        Notebook Profissional Avell W155 MX Intel Core i7 16GB (GeForce MX150) 1TB 15.6 FullHD
      </a>
    </h1>
    <div className=\\"info\\">
      <a target=\\"_blank\\" href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
        R$ 5.333,20
      </a>
    </div>
    <div className=\\"footer\\">
      <div className=\\"expires\\">
        <a target=\\"_blank\\" href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
          Valido até: 09/07/2018 2:59
        </a>
      </div>
      <div className=\\"social\\">
        <Button className=\\"copy\\" onClick={[undefined]} loading={false} iconRight=\\"insert_link\\" color=\\"primary\\" modifier=\\"outline\\" disabled={false} size=\\"medium\\" type=\\"button\\" active={false}>
          Gerar Link
        </Button>
        <Svg onClick={[Function]} className=\\"facebook\\" align=\\"top\\" width={26} height={26} src=\\"icon/facebook-square\\" />
        <Svg onClick={[Function]} className=\\"twitter\\" align=\\"top\\" width={26} height={26} src=\\"icon/twitter-square\\" />
      </div>
    </div>
  </Symbol(react.fragment)>
</section>"
`;

exports[`Product component Variations Product with no info 1`] = `""`;

exports[`Product component Variations Type card renders Panel 1`] = `
"<Panel title={{...}} contentClassName=\\"wrapper card\\" size=\\"no-padding\\" data={{...}} stage=\\"generate\\" onCopy={[Function: onCopy]} loading={false} accordion={false} open={true} highlight=\\"shadow\\">
  <Symbol(react.fragment)>
    <div className=\\"tag\\">
      <Svg width={48} height={48} src=\\"logo/acom\\" align=\\"middle\\" />
      <Svg className=\\"tagHighlight\\" width={32} height={32} src=\\"flame\\" align=\\"middle\\" />
    </div>
    <div className=\\"imgWrapper\\">
      <a target=\\"_blank\\" href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
        <img className=\\"imgCustom\\" src=\\"https://images-americanas.b2w.io/produtos/01/00/sku/33446/6/33446652_4GG.jpg\\" alt=\\"Notebook Profissional Avell W155 MX Intel Core i7 16GB (GeForce MX150) 1TB 15.6 FullHD\\" />
      </a>
    </div>
    <h1 className=\\"name\\">
      <a href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
        Notebook Profissional Avell W155 MX Intel Core i7 16GB (GeForce MX150) 1TB 15.6 FullHD
      </a>
    </h1>
    <div className=\\"info\\">
      <a target=\\"_blank\\" href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
        R$ 5.333,20
      </a>
    </div>
    <div className=\\"footer\\">
      <div className=\\"expires\\">
        <a target=\\"_blank\\" href=\\"https://www.americanas.com.br/produto/33446653/notebook-profissional-avell-w155-mx-intel-core-i7-16gb-geforce-mx150-1tb-15-6-fullhd\\">
          Valido até: 09/07/2018 2:59
        </a>
      </div>
      <div className=\\"social\\">
        <Button className=\\"copy\\" onClick={[undefined]} loading={false} iconRight=\\"insert_link\\" color=\\"primary\\" modifier=\\"outline\\" disabled={false} size=\\"medium\\" type=\\"button\\" active={false}>
          Gerar Link
        </Button>
        <Svg onClick={[Function]} className=\\"facebook\\" align=\\"top\\" width={26} height={26} src=\\"icon/facebook-square\\" />
        <Svg onClick={[Function]} className=\\"twitter\\" align=\\"top\\" width={26} height={26} src=\\"icon/twitter-square\\" />
      </div>
    </div>
  </Symbol(react.fragment)>
</Panel>"
`;
